### DISCUSSION TITLE
hardware stability with the USBHost

### DISCUSSION THREAD

**Discussion by MichaelPNolan** [marcel-licence/esp32_basic_synth#49] [Ideas]
Just a mention here that I had unreliable results from the MAX3421E based host. I had set the vbus to 5v by modifying the pcb as various people suggested (not sure if that was wise). I found that lowering the voltage to 4.35v the communications stopped getting stuck at waiting for reset. I don't know if this is really helping or if it is random.

This is with a Novation Launchpad mini v2 - it is known to run at lower power such as OTG ipad usb power. It has a low power mode. I know others had some problem until they ensured there was 3.3v communications which makes sense if the chip is running at 3.3v (and assumedly SPI runs at that). I'm using a basic ESP32 with no dac.

**Comment by marcel-licence** [comment]
Hm it looks like there are also different version of MAX3421E based breakout boards. The 5V should be stable. I had also one device which didn't want to start when connected to the MAX. The Launchpad Pro (I'm using) also showed some issues. The final setup seems to work more reliable. Maybe there are some floating signal lines (reset?). The MIDI part of the library in arduino wasn't updated for a long time. If you can get a cheap DAC you would have a lot different experience regarding the sound quality =)

**Reply by MichaelPNolan** [reply]
Yes, i changed the power supply and I think the 3.3v is important for a better communication. Something odd with this regulator the 3.3v rail on esp32 was a bit higher voltage. So I supplied the USB Host it's own 3.3v and it worked again. Every time I think i solved it then after a cold start the issue is back ... so i'll look at changing out some wires or something.

My DAC sounds pretty fine - I'm testing it with internet radio with essentially the same I2s library. I'm a guitar player - I don't have a master keyboard - just the launchpad. I'll have to aquire one, i can play very simplistic things on keys. My DAC takes 5v+ supply and has 3 regulators for 3.3v to the IC. Doesn't seem to create any issue the communications is still 3.3v

**Comment by MichaelPNolan** [comment]
Ok - this was old and lived in a box forever. This Novation Launchpad  has died now - so maybe this is why it was flakey - it was playing up and one day it failed. Went to pull too much current and cascading fail. RIP. So now to buy. 

So ... like a fool i went on and used my phone as the USB source ... but with a more safe power source. All is ok, it has a midi output mode.

I spun up a midi controller in android and basically that works - so I don't have to wait for anything except that with no analogue midi controls I just get this horrible bitcrunched sound that sucks ... so have to fix that. I disabled ADC controller ... still horrid but more tonality there. So definitely working.
I'm starting to wonder if I can have a webserver with controls ... that would mess with the CPU power / resources for this sketch?

I think a pair of buttons that scroll through patches for the parameters ADC does control. I guess its time to build the hardware aspects of this project - still waiting for multiplexer.
